MrtCore/console_unpackaged_app/console_unpackaged_app.cpp

```diff
@@ -0,0 +1,114 @@
+        return hr;
+    }
+
+    return S_OK;
+}
+
+ResourceManager InitMrtCore()
+{
+    // Required for C++/WinRT. This call associates this thread with an apartment and initializes COM runtime.
+    init_apartment();
+
+    // Create a resource manager using the resources index (PRI file) generated during build.
+    auto factory = winrt::get_activation_factory<ResourceManager, IResourceManagerFactory>();
+    return factory.CreateInstance(L"console_unpackaged_app.pri");
+}
+
+int wmain(int argc, wchar_t* argv[])
+{
+    // Print usage help.
+    if ((argc < 2) || (wcscmp(argv[1], L"--help") == 0) || (wcscmp(argv[1], L"-?") == 0))
+    {
+        std::cout << "Usage: console_unpackaged_app.exe [options] [mode]\n"
+            "options:\n"
+            "  -?, --help = Display help\n"
+            "mode:\n"
+            "  Mode for app run. Can be...\n"
+            "       Default or default\n"
+            "       Override or override\n"
+            "       Fallback or fallback\n"
+            "  Default and override retrieve a sample string from string resource files.\n"
+            "  For the override case, this sample uses the German language.\n"
+            "  Fallback corresponds to the resource-not-found case, where we fallback to a legacy resource loader.\n"
+            "\n"
+            "Examples:\n"
+            "  Get the sample string for the default resource context\n"
+            "    console_unpackaged_app.exe default\n"
+            "  Get the sample string for the override resource context (sample uses the German language for the override context)\n"
+            "    console_unpackaged_app.exe override\n"
+            "  Get the sample string for the resource-not-found fallback case\n"
+            "    console_unpackaged_app.exe fallback\n";
+        return 1;
+    }
+
+    // Initialize dynamic dependencies so we can consume the Project Reunion APIs in the Project Reunion framework package from this unpackaged app. 
+    HRESULT mddInitHr = MddBootstrapInitializeWrapper();
+    if (FAILED(mddInitHr))
+    {
+        return 1;
+    }
+
+    ResourceManager manager = InitMrtCore();
+    manager.ResourceNotFound([](ResourceManager const&, ResourceNotFoundEventArgs const& args)
+        {
+            // There could be a resource in a legacy resource file that we retrieve using the corresponding legacy resource loader. For example, the C#
+            // MRT Core sample apps in this repo use the .NET resource loader as their fallback. Instead, we just hardcode a resource value here.
+            if (args.Name() == L"Resources/LegacyString")
+            {
+                auto candidate = ResourceCandidate(ResourceCandidateKind::String, L"Legacy Sample");
+                args.SetResolvedCandidate(candidate);
+            }
+        });
+
+    if ((wcscmp(argv[1], L"Default") == 0) || (wcscmp(argv[1], L"default") == 0))

```

Comment:
>wcscmp [](http://example.com/codeflow?start=9&length=6)

_wcsicmp

Reply:
I was wondering why, and found this: https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/stricmp-wcsicmp?view=msvc-160
> The Microsoft-specific function names stricmp and wcsicmp are deprecated aliases for the _stricmp and _wcsicmp functions.